### PR TITLE
chore: upgrade StyleLint version to 15

### PR DESCRIPTION
**Pull request by oguemon**
Upgrade StyleLint version from 14 to 15.

Ref: [Migrating to 15.0.0 | Stylelint](https://stylelint.io/migration-guide/to-15)

### CODE REVIEW

File: styled-components.js [hunk 1]
```diff
@@ -11,12 +11,6 @@ module.exports = {
     // OK
     // background-image: url(${toStaticUrl('/img/foo.png')});
     "function-url-quotes": null,
-    // This rule cannot recognize CSS functions in template literals
-    // Error
-    // transform: ${({ rotate }) => `rotate(${rotate}deg)`};
-    // OK
-    // transform: ${({ rotate }) => `rotate(${rotate}deg) `};
-    "function-whitespace-after": null,
```
Comment (oguemon) [inline styled-components.js:19]: memo
`function-whitespace-after` is now deprecated and removed from `stylelint-config-standard`.

https://github.com/stylelint/stylelint-config-standard/compare/29.0.0...30.0.0#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L93

File: index.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  extends: ["stylelint-config-standard", "stylelint-config-prettier"],
```
Comment (oguemon) [inline index.js:2]: memo
Ref: https://stylelint.io/migration-guide/to-15#deprecated-stylistic-rules.

File: package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "test": "jest"
   },
   "engines": {
-    "node": ">=12"
+    "node": ">=14.13.1"
```
Comment (oguemon) [inline package.json:18]: memo
Removed support for Node.js 12.
https://stylelint.io/migration-guide/to-15#removed-support-for-nodejs-12

File: styled-components.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   extends: ["./index.js", "stylelint-config-styled-components"],
-  customSyntax: "@stylelint/postcss-css-in-js",
```
Comment (oguemon) [inline styled-components.js:3]: memo
[@stylelint/postcss-css-in-js](https://www.npmjs.com/package/@stylelint/postcss-css-in-js) is deprecated, so I use [postcss-styled-syntax](https://www.npmjs.com/package/postcss-styled-syntax) alternatively.

https://stylelint.io/migration-guide/to-15#removed-processors-configuration-property

File: styled-components.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
...
+  customSyntax: "postcss-styled-syntax",
```
Comment (oguemon) [inline styled-components.js:3]: memo
[@stylelint/postcss-css-in-js](https://www.npmjs.com/package/@stylelint/postcss-css-in-js) is deprecated, so I use [postcss-styled-syntax](https://www.npmjs.com/package/postcss-styled-syntax) alternatively.

https://stylelint.io/migration-guide/to-15#removed-processors-configuration-property

File: index.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
...
+  extends: ["stylelint-config-standard"],
```
Comment (oguemon) [inline index.js:2]: memo
Ref: https://stylelint.io/migration-guide/to-15#deprecated-stylistic-rules.

**Comment by koba04** [comment]
@oguemon 
Thank you!
Could you update the Node.js version on GitHub Actions? Let's remove `12.x` and add `18.x` instead.

https://github.com/cybozu/stylelint-config/blob/main/.github/workflows/test.yml#L17

**Review by koba04 — APPROVED** [review]
👍 LGTM!!! Great work 👍

**Comment by koba04** [comment]
@oguemon Could you fix the failing test?

**Comment by oguemon** [comment]
@koba04
I fixed the failing test on Node.js 14! (Add Package https://github.com/cybozu/stylelint-config/pull/27/commits/10bc6c2f6bd49473bb6471030f046c1b9eb3ff04)

And I updated `actions/setup-node@v2` -> `@v3` (https://github.com/cybozu/stylelint-config/pull/27/commits/a90efef8b8054c53227b16275b025246968d8a07)

Prease check these changes!

**Pull request merged**